addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +113,144 @@ def _get_duration_from_tracking(self, trackings):
             previous_date = tracking['create_date']
 
         return json
+
+    """
+    Rotting logic
+
+    The rotting feature enables resources to mark themselves as stale if enough time has passed since they were last updated
+    by an user.
+    To enable this behavior, the following must be done:
+    - The model pointed to by _track_duration_field must have a "rotting_threshold_days" integer field, representing the
+    number of days before the resource is considered stale:
+        on the model pointed to by _track_duraton_field:
+            rotting_threshold_days = fields.Integer('Days to rot', default=0)
+
+    - The inheriting model must have a date_last_stage_update field, tracking the date of the last time
+    the tracked field was updated
+
+    - The following methods need to be overriden:
+      - _get_rotting_depends_fields() must be overriden to include the rotting_threshold_days field, as well as
+        any field that should affect the ability of a resource to rot (for example: a field that would mark a task as done):
+            def _get_rotting_depends_fields(self):
+                return super()._get_rotting_depends_fields() + ['won_status', 'type']
+
+      - _get_rotting_domain() must be overriden to include conditions that must be met so that the field can be considered rotting:
+            def _get_rotting_domain(self):
+                return super()._get_rotting_domain() + [
+                    ('won_status', '=', 'pending'),
+                    ('type', '=', 'opportunity')
+                ]
+
+        Note that the fields present in _get_rotting_domain() are the same as the ones present in _get_rotting_depends_fields()
+
+    - The is_rotting, rotting_days fields need to be added to the relevant views
```

Comment:
This is globally a bit irrelevant, as it is standard in odoo to add required fields in view. Maybe a note for kanban widget, but keep it short :)